core.spec.md

```diff
@@ -0,0 +1,501 @@
+
+# Processing Schemas
+
+```mermaid diagram
+graph LR
+  schema(["📄  Input Schema"]):::file-->proc("🤖 &nbsp;Processor")
+  proc-->output(["📄  Output Schema"]):::file
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common use case is that of a processor which consumes a valid input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - SHOULD pass through {@core} directives which reference unknown feature URLs
+  - SHOULD pass through prefixed directives, types, and other schema elements  
+  - SHOULD pass through directives which are not [associated with](#AssignFeatures) a {@core} feature
+
+Processors MAY accept configuration which overrides these default behaviors.
+
+Additionally, processors which prepare the schema for final public consumption MAY choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+# Validations &amp; Algorithms
+
+## Bootstrapping (validates Has Core Feature)
+
+Determine the name of the core specification within the document *or* fail the *Has Core Feature* validation.
+
+It is possible to [rename the core feature](#sec-Renaming-core-itself) within a document. This process determines the actual name for the core feature if one is present.
+
+Bootstrap(document) :
+1. For each directive {d} on the SchemaDefinition within {document},
+  1. If {d} has a [`feature:`](#@core/feature) argument whose value is "https://lib.apollo.dev/core/v0.1", *and either*:
+    - &#8230;{d} has an [`as:`](#@core/as) argument whose value is equal to {d}'s name
+    - &#8230;*or* {d} does not have an [`as:`](#@core/as) argument and {d}'s name is `core`
+    - *then* **Return** `coreName` = {d}'s name
+- If no matching directive was found, the ***Has Core Feature* validation fails**.
+
+## Feature Collection (validates Name Uniqueness)
+
+Collect a map of ({featureName}: `String`) -> `Directive`, where `Directive` is a {@core} Directive which introduces the feature named {featureName} into the document.
+
+CollectFeatures(document) :
+  - Let {coreName} be the name of the core feature found via {Bootstrap(document)}
+  - Let {features} be a map of {featureName}: `String` -> `Directive`, initially empty.
+  - For each directive {d} named `coreName` on the SchemaDefinition within {document},
+    - Let {name} be the spec's [name](#sec-Prefixing) as specified by the directive's [`as:`](#@core/as) argument or, if the argument is not present, the default name from the [feature url](#@core/feature).
```

Comment:
SUBSTANTIVE: Should be explicit that this feature url is on the `feature` argument.

What happens if there is no `feature` argument, if a non-string is passed, or if the string does not parse as an URL with appropriate properties for name and version?

(In general, do we want error handling to be "if anything fails then everything fails" or to give partial results?)

Reply:
Fixed, we have the concept of warnings and validation failures now, to distinguish cases where we want to halt processing from those where we can continue to extract partial data.